ci/docker.sh

```diff
@@ -0,0 +1,70 @@
+#!/bin/bash
+set -eEuo pipefail
+
+workdir="$( cd "$( dirname "${BASH_SOURCE[0]}" )/.." &> /dev/null && pwd )"
+docker_image_build_name="${DOCKER_IMAGE_BUILD_NAME:-nh-node}"
+docker_hub_org="${DOCKER_HUB_ORG:-horizenlabs}"
+docker_hub_username="${DOCKER_HUB_USERNAME:-}"
+docker_hub_token="${DOCKER_HUB_TOKEN:-}"
+is_a_release="${IS_A_RELEASE:-false}"
+prod_release="${PROD_RELEASE:-false}"
+dev_release="${DEV_RELEASE:-false}"
+test_release="${TEST_RELEASE:-false}"
+github_ref_name="${GITHUB_REF_NAME:-}"
+common_file_location="${COMMON_FILE_LOCATION:-not-set}"
+docker_file_path='docker/dockerfiles/hl-node.Dockerfile'
+
+# Requirement
+if ! [ -f "${common_file_location}" ]; then
+  echo -e "\n\033[1;31mERROR: ${common_file_location} file is missing !!!  Exiting ...\033[0m\n"
+  exit 1
+else
+  # shellcheck disable=SC1090
+  source "${common_file_location}"
+fi
+
+
+####
+# Main
+####
+if [ -z "${docker_hub_token:-}" ]; then
+  fn_die "ERROR: DOCKER_HUB_TOKEN variable is not set. Exiting ..."
+fi
+
+if [ -z "${docker_hub_username:-}" ]; then
+  fn_die "ERROR: DOCKER_HUB_USERNAME variable is not set. Exiting ..."
+fi
+
+docker_tag=""
+if [ "${is_a_release}" = "true" ]; then
+  docker_tag="${github_ref_name}"
+fi
+
+# Building and publishing docker image
+if [ -n "${docker_tag:-}" ]; then
+  log italic green "=== Building Docker image: ${docker_hub_org}/${docker_image_build_name}:${docker_tag} ==="
+  #docker build -f "${docker_file_path}" -t "${docker_hub_org}/${docker_image_build_name}:${docker_tag}" .
```

Comment:
Please uncomment `docker build`, `docker tag` and `docker push` commands.

Reply:
That was the plan for the last commit once everything else is reviewed and approved. The plan is not to build docker image everytime during the implementation